.github/workflows/ui-test-execution.yml

```diff
@@ -0,0 +1,142 @@
+name: UI Automated Tests on MacOS 🧪
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build-app:
+    runs-on: macos-latest
+
+    steps:
+      - name: Checkout Repo 🔖
+        uses: actions/checkout@v3
+
+      - name: Set up cargo cache 🛠️
+        uses: actions/cache@v3
+        continue-on-error: false
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-cargo-
+
+      - name: Install Rust 💿
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+
+      - name: Install Protobuf 💿
```

Comment:
Not sure how it will work with a mac target but you could use `arduino/setup-protoc@v1` action for installing protobuf. 

See https://github.com/Satellite-im/Warp/blob/main/.github/workflows/warp_dart_update.yml#L16-L17 

Not a blocker though, especially if the gh action script doesnt work with a mac target

Reply:
Thanks @dariusc93 ! Trying out in a new commit the `arduino/setup-protoc@v1` action to install protobuf. Could you let me know how can I improve the cargo cache step or if it is better to remove it? I think the action should work and build the app even without this step

Reply:
The caching part can stay.

Reply:
Thanks!